src/compiler/mod.rs

```diff
@@ -100,31 +130,86 @@ impl<'a> Compiler<'a> {
         }
     }
 
+    fn push_to_datatable(&mut self, value: PactType<'a>) -> Result<(), CompileErr> {
+        self.data_table.push(value);
+        if self.data_table.len() > MAX_ENTRIES {
+            Err(CompileErr::DataTableFull)
+        } else {
+            Ok(())
+        }
+    }
+
     /// Compile an assertion AST node
     fn compile_assertion(&mut self, assertion: &'a ast::Assertion) -> Result<(), CompileErr> {
-        let comparator_op = compile_comparator(&assertion.2)?;
-        self.bytecode.push(comparator_op.into());
-
-        let lhs_load_op = self.compile_subject(&assertion.0)?;
-        match lhs_load_op {
-            OpCode::LD_INPUT(index) | OpCode::LD_USER(index) => {
-                self.bytecode.push(lhs_load_op.into());
-                self.bytecode.push(index);
-            }
-            _ => panic!("unreachable"),
+        let lhs_load = self.compile_subject(&assertion.0)?;
```

Comment:
would be clearer if we used named fields instead of .0 and .1

Reply:
done!